lib/config/fileconf.go

```diff
@@ -1108,6 +1109,13 @@ func (ssh *SSH) X11ServerConfig() (*x11.ServerConfig, error) {
 	return cfg, nil
 }
 
+type Discovery struct {
+	Service `yaml:",inline"`
+
+	// AWSMatchers are used to match EC2 instances
+	AWSMatchers []AWSEC2Matcher `yaml:"aws,omitempty"`
```

Comment:
As discussed offline, the current structure is ok since we're using "types" field to filter by types.

However, since we'll be using this section for other resource types as well, not just EC2 instances, I don't think `AWSEC2Matcher` name is really fitting. Can we rename it to just `AWSMatcher`?